tests/test_dimension.py

```diff
@@ -830,6 +831,76 @@ def test_implicit_dims(self):
 
         assert np.all(f.data == F)
 
+    def test_condition_lowering(self):
+        """
+        Test the lowering of a ConditionalDimension's condition.
+        This test makes an Operator that should indexify and lower the condition
+        passed in the Conditional Dimension
+        """
+
+        grid = Grid(shape=(3, 3))  # Define grid
+
+        # Define Function
+        g = Function(name='g', shape=grid.shape, dimensions=grid.dimensions)
+        g.data[:] = 1
+        x, y = grid.dimensions
+        ci = ConditionalDimension(name='ci', parent=y, condition=Le(g, 1))
+
+        f = Function(name='f', shape=grid.shape, dimensions=(x, ci))
+        Operator(Eq(f, g)).apply()
+        assert np.all(f.data == 1)
+
+    @pytest.mark.parametrize('setup_rel, rhs, c1, c2, c3, c4', [
+        # Relation, RHS, c1 to c4 used as indexes in assert
+        (Lt, 3, 2, 4, 4, -1), (Le, 2, 2, 4, 4, -1), (Ge, 3, 4, 6, 1, 4),
+        (Gt, 2, 4, 6, 1, 4), (Ne, 5, 2, 6, 1, 2)
+    ])
+    def test_relational_classes(self, setup_rel, rhs, c1, c2, c3, c4):
+        """
+        Test ConditionalDimension using conditions constructed with overriden
+        SymPy relation classes.
+        """
+
+        class InnerDomain(SubDomain):
+            name = 'inner'
+
+            def define(self, dimensions):
+                return {d: ('middle', 2, 2) for d in dimensions}
+
+        inner_domain = InnerDomain()
+        grid = Grid(shape=(8, 8), subdomains=(inner_domain,))
+        g = Function(name='g', shape=grid.shape, dimensions=grid.dimensions)
+
+        g.data[:4, :4] = 1
+        g.data[4:, :4] = 2
+        g.data[4:, 4:] = 3
+        g.data[:4, 4:] = 4
+
+        xi, yi = grid.subdomains['inner'].dimensions
+
+        cond = setup_rel(g, rhs, subdomain=grid.subdomains['inner'])
+        ci = ConditionalDimension(name='ci', parent=yi, condition=cond)
+        f = Function(name='f', shape=grid.shape, dimensions=(xi, ci))
+
+        Eq1 = Eq(f, g)
```

Comment:
variables should basically never start with capital letters

```
eqns = [Eq(...), Eq(...)]
```

since they're tiny

Reply:
Fixed